src/components/TeamGenerator/index.tsx

```diff
@@ -1,30 +1,64 @@
-import { nanoid } from 'nanoid';
 import React, { useState } from 'react';
-import { Button, Icon, Input, Label, Modal, TextArea } from 'semantic-ui-react';
+import { Divider, Button, Text, Checkbox, Group } from '@mantine/core';
 import { getCommaSeparatedStringWithAndBeforeTheLastItem } from '../../helpers';
-import { Team } from '../../types';
+import { FormTextArea, FormInput } from '../FormInputs';
+import { useForm } from 'react-hook-form';
+import Icon from '../../components/Icon';
 import styles from './styles.module.css';
+import { useStore } from '../../useStore';
 
-interface Props {
-  okCallback: Function;
-  numOfTeams: number;
-  hideModal: Function;
-}
-
-export default function TeamGenerator({ okCallback, numOfTeams, hideModal }: Props) {
-  const [teamCount, setTeamCount] = useState<number | ''>(numOfTeams);
-  const [teams, setTeams] = useState<Team[]>([]);
-  const [playerNames, setPlayerNames] = useState('');
+export default function TeamGenerator({ createTeams, ...rest }) {
+  const [editTeams, setEditTeams] = useState(!!rest.players);
+  const [showTeams, setShowTeams] = useState(!!rest.players);
+  const [players, setPlayers] = useState<string[]>(rest.teams.map((x) => x.players));
+  const { showAlert } = useStore();
+  const [, setRefresh] = useState(0);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    getValues,
+  } = useForm({
+    defaultValues: {
+      playerNames: rest.players,
+      teamCount: rest.teamCount,
+    },
+  });
+  const teamsForm = useForm({
+    defaultValues: {
+      teams:
+        rest.teams.length > 0
+          ? rest.teams
+              .map((x) => x.name)
+              .join(',')
+              .replaceAll(',', '\n')
+          : '',
+    },
+  });
+  const teamList = teamsForm
+    .getValues('teams')
+    .split('\n')
+    .filter((x) => !!x);
 
-  function generateTeams() {
-    const totalTeams = teamCount || numOfTeams || 2;
-    const teamsPlayers = new Array<string[]>(totalTeams);
+  function generateTeams(data) {
+    const { teamCount, playerNames } = data;
+    const count = parseInt(teamCount);
+    const teamsPlayers = new Array<string[]>(count);
 
     let validPlayerNames = playerNames
       .split('\n')
       .map((name) => name.trim())
       .filter((name) => !!name);
```

Comment:
We should create a helper function like `getValidValuesFromColumns` which does `.split('\n')
      .map((x) => x.trim())
      .filter((x) => !!x);` and reuse it here & above.